dashboard/app/models/user_script.rb

```diff
@@ -36,4 +38,17 @@ def empty?
     # a user script is empty if there is no progress
     started_at.nil? && completed_at.nil? && assigned_at.nil? && last_progress_at.nil?
   end
+
+  def check_plc_task_assignment_updating
+    if check_completed? && self.script.pd
```

Comment:
It should be, that is what it is for. Since you only fire this callback on UserScript, the update to completed_at is the event that you are already depending on.

When last_progress_at is updated, completed_at is supposed to be updated (if necessary) via a call to check_completed?: https://github.com/code-dot-org/code-dot-org/blob/4af265701340ec08e2e54e5ff589a482ff70d9af/dashboard/app/models/user.rb#L791